packages/docs/build/prerender.js

```diff
@@ -0,0 +1,135 @@
+os = require('os')
+const fs = require('fs')
+const path = require('path')
+const util = require('util')
+const generateRoutes = require('./generate-routes')
+const { DynamicPool } = require('node-worker-threads-pool')
+
+const threads = os.cpus().length
+const resolve = file => path.resolve(__dirname, file)
+
+const pool = new DynamicPool(threads)
+
+function log (msg) {
+  console.info('\n')
+  console.info(msg)
+  console.info('\n')
+}
+
+function chunk (arr, chunkSize) {
+  const chunks = []
+
+  for (let i = 0; i < arr.length; i += chunkSize) {
+    chunks.push(arr.slice(i , i + chunkSize))
+  }
+
+  return chunks
+}
+
+function promise (cb) {
+  return new Promise(resolve => {
+    let value
+
+    try {
+      value = cb()
+    } catch (err) {
+      log(err)
+
+      process.exit(1)
+    }
+
+    resolve(value)
+  })
+}
+
+function readFile (file) {
+  const read = util.promisify(fs.readFile)
+
+  return promise(() => read(resolve(file), 'utf-8'))
+}
+
+async function readTemplate () {
+  return readFile('../src/index.template.html')
+}
+
+async function readBundle () {
+  return JSON.parse(await readFile('../dist/vue-ssr-server-bundle.json'))
+}
+
+async function readManifest () {
+  return JSON.parse(await readFile('../dist/vue-ssr-server-bundle.json'))
+}
+
+async function run () {
+  const routes = await promise(generateRoutes).catch(log)
```

Comment:
These were async to not block webpack, you can just replace it with `readFileSync` now that it's standalone.

Reply:
Also promise() doesn't reject so why the .catch()?

Reply:
And generateRoutes is already a promise